ktor-http/common/src/io/ktor/http/URLBuilder.kt

```diff
@@ -23,15 +23,15 @@ public const val DEFAULT_PORT: Int = 0
  * @property trailingQuery keep a trailing question character even if there are no query parameters
  */
 public class URLBuilder(
-    public var protocol: URLProtocol = URLProtocol.HTTP,
-    public var host: String = "localhost",
-    public var port: Int = DEFAULT_PORT,
-    user: String? = null,
-    password: String? = null,
-    pathSegments: List<String> = mutableListOf(""),
-    parameters: Parameters = Parameters.Empty,
-    fragment: String = "",
-    public var trailingQuery: Boolean = false
+    public var protocol: URLProtocol? = null,
+    public var host: String? = null,
+    public var port: Int? = null,
+    public var user: String? = null,
```

Comment:
It's essential that these parameters are not properties. Otherwise encode/decode process gets broken.